framework/source/class/qx/util/ResourceManager.js

```diff
@@ -285,9 +285,10 @@ qx.Class.define("qx.util.ResourceManager",
 
           var href;
           //first check if there is base url set
-          var baseElements = qx.bom.Selector.query("base", document);
-          if (baseElements.length > 0) {
-            href = baseElements[0].href;
+          if (typeof window == "object" &&
```

Comment:
Please can we make this a `qx.core.Environment` check?  The reason is that we already fake global `window` when generating server apps so that the global/default object *is* `window` and therefore the `typeof window == "object"` will be true, but I doubt if this is documented anywhere.

Also, recently I started faking `window.document` in server apps generated by qxcompiler - ironically, one reason was because of this `qx.bom.Selector` that @derrell is fixing, but AFAICR we also have instances where third party code (eg the Bluebirdjs implementation embedded in `qx.Promise`) will expect `document` to exist and have various methods.  

It is a fine line as to how far we would go when mocking, but it would be helpful if the detection code was isolated into a single environment check.

Reply:
@johnspackman I see your point. We should maybe enhance `qx.bom.client.Runtime` to not just offer `runtime.name` but also `runtime.document`, `runtime.window` or similar, which should return true if window and document are available as global objects. 

Maybe guarded with some specific tests to be able to distinguish them from fake objects.

A possible test could be to call `document.getElementsByTagName('somenonexistingtagname')` which should succeed and return an empty array.

Reply:
or
```javascript
typeof document === "object" && document instanceof Document
```

Reply:
For the moment lets not have `runtime.window` because there is lots of code which assumes that there is a `window` object already, but `runtime.document` would be a good idea.  The boot loader for non-browser can always emit some undocumented values (such as `qx.$$loader.isFakeDocument=true` that allow the `runtime.document` implementation to understand what's going on